### PR TITLE
Add `setup` tag for config structs

### PR DESCRIPTION
**Pull request by mniewrzal**
This PR will add `setup:"true"` tag for config structures. Struct or field with such tag will be visible only for setup command and won't be saved in config file.

### CODE REVIEW

File: pkg/cfgstruct/bind.go [hunk 1]
```diff
@@ -147,10 +164,26 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value,
...
 
+func setSetupAnnotation(flagset interface{}, name string) {
+	flags, ok := flagset.(*pflag.FlagSet)
+	if !ok {
+		return
+	}
+
+	err := flags.SetAnnotation(name, "setup", []string{"true"})
+	if err != nil {
+		fmt.Println("unable to set annotation:", err)
```
Comment (coyle) [inline pkg/cfgstruct/bind.go]: should we use a logger?

Comment (mniewrzal) [reply pkg/cfgstruct/bind.go]: I think we aren't using logger in commands, at least for now. tbh I'm not sure if this should be logged at all.

File: pkg/cfgstruct/bind.go [hunk 2]
```diff
@@ -86,24 +99,28 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value,
...
-				bindConfig(flags, flagname+".", fieldval, vars)
+				bindConfig(flags, flagname+".", fieldval, vars, setupCommand, setup)
 			}
 		case reflect.Array, reflect.Slice:
 			digits := len(fmt.Sprint(fieldval.Len()))
 			for j := 0; j < fieldval.Len(); j++ {
 				padding := strings.Repeat("0", digits-len(fmt.Sprint(j)))
 				bindConfig(flags, fmt.Sprintf("%s.%s%d.", flagname, padding, j),
-					fieldval.Index(j), vars)
+					fieldval.Index(j), vars, setupCommand, setup)
```
[the remark this answers is not in the document]
Comment (mniewrzal) [reply pkg/cfgstruct/bind.go]: done

File: pkg/cfgstruct/bind.go [hunk 3]
```diff
@@ -55,15 +68,15 @@ func Bind(flags FlagSet, config interface{}, opts ...BindOpt) {
...
+	bindConfig(flags, "", reflect.ValueOf(config).Elem(), vars, setupCommand, false)
 }
 
 var (
 	whitespace = regexp.MustCompile(`\s+`)
 )
 
 func bindConfig(flags FlagSet, prefix string, val reflect.Value,
-	vars map[string]confVar) {
+	vars map[string]confVar, setupCommand, setupStruct bool) {
```
[the remark this answers is not in the document]
Comment (mniewrzal) [reply pkg/cfgstruct/bind.go]: done

File: pkg/cfgstruct/bind.go (hunk 1 above)
Comment (mniewrzal) [reply pkg/cfgstruct/bind.go]: done

File: pkg/cfgstruct/bind.go [hunk 4]
```diff
@@ -86,24 +99,28 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value,
 		field := typ.Field(i)
 		fieldval := val.Field(i)
 		flagname := prefix + hyphenate(snakeCase(field.Name))
+		tag := reflect.StructTag(whitespace.ReplaceAllString(string(field.Tag), " "))
```
[the remark this answers is not in the document]
Comment (mniewrzal) [reply pkg/cfgstruct/bind.go]: done

File: pkg/cfgstruct/bind.go [hunk 5]
```diff
@@ -86,24 +99,28 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value,
...
+		setup := (tag.Get("setup") == "true") || setupStruct
```
[the remark this answers is not in the document]
Comment (mniewrzal) [reply pkg/cfgstruct/bind.go]: done

**Pull request merged**